test/integration/dredd-test.coffee

```diff
@@ -221,6 +222,65 @@ describe "Dredd class Integration", () ->
       it 'prints out ok', ->
         assert.equal exitStatus, 0
 
+    describe "when using reporter -r apiary and the server isn't running", () ->
+      server = null
+      server2 = null
+      receivedRequest = null
+      exitStatus = null
+
+      before (done) ->
+        cmd =
+          options:
+            path: ["./test/fixtures/single-get.apib"]
+            reporter: ['apiary']
+          custom:
+            apiaryReporterEnv:
+              APIARY_API_URL: "http://127.0.0.1:#{PORT+1}"
+              DREDD_REST_DEBUG: '1'
+
+        apiary = express()
+
+        apiary.use bodyParser.json(size:'5mb')
+
+        apiary.post '/apis/*', (req, res) ->
+          if req.body and req.url.indexOf('/tests/steps') > -1
+            receivedRequest ?= clone(req.body)
+          res.type('json')
+          res.status(201).send
+            _id: '1234_id'
+            testRunId: '6789_testRunId'
+            reportUrl: 'http://url.me/test/run/1234_id'
+
+        apiary.all '*', (req, res) ->
+          res.type 'json'
+          res.send {}
+
+        server2 = apiary.listen (PORT+1), ->
+
+          # Comment out this timeout to enable race condition bug
+          # Event loop get stuck or idle without this
```

Comment:
maybe we could have sth like `wakeUpEventLoop()` function shared among all these tests, which will be documented by the comment on one place and the implementation will be at one place and it will be clearly marked as workaround for a bug?